crates/ra_ide_api/src/goto_definition.rs

```diff
@@ -216,6 +216,29 @@ mod tests {
         );
     }
 
+    #[test]
+    fn goto_definition_works_for_macros_from_other_crates() {
+        covers!(goto_definition_works_for_macros);
+        check_goto(
+            "
+            //- /lib.rs
+            use foo::foo;
+            fn bar() {
+                <|>foo!();
+            }
+
+            //- /foo/lib.rs
```

Comment:
Hm, do we ever specify that `/foo/lib.rs` is an extern crate?

Reply:
Yes if i understand correctly :)

https://github.com/rust-analyzer/rust-analyzer/blob/7a1cae59acf72f821343b2ba10ef69fb92a5b952/crates/ra_ide_api/src/mock_analysis.rs#L80-L100